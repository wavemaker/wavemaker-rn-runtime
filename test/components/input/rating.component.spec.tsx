import React, { ReactNode } from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import WmRating from '@wavemaker/app-rn-runtime/components/input/rating/rating.component';

describe('Test Rating component', () => {
    test('Check validity of sample component', () => {
      const tree = renderer.create(<WmRating name="test_Rating"/>).toJSON();
      expect(tree).toMatchSnapshot();
    });
});