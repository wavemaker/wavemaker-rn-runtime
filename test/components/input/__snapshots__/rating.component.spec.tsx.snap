// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WmRating renders correctly with default props 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "flexDirection": "row",
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <View
    accessibilityLabel=""
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
        }
      }
      testID="non_animatableView"
    >
      <View>
        <View>
          <Text
            size={32}
            style={
              [
                {
                  "color": "#aaaaaa",
                },
                {
                  "paddingLeft": 0,
                  "paddingRight": 8,
                },
                {
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          >
            star-border
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel=""
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
        }
      }
      testID="non_animatableView"
    >
      <View>
        <View>
          <Text
            size={32}
            style={
              [
                {
                  "color": "#aaaaaa",
                },
                {
                  "paddingLeft": 0,
                  "paddingRight": 8,
                },
                {
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          >
            star-border
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel=""
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
        }
      }
      testID="non_animatableView"
    >
      <View>
        <View>
          <Text
            size={32}
            style={
              [
                {
                  "color": "#aaaaaa",
                },
                {
                  "paddingLeft": 0,
                  "paddingRight": 8,
                },
                {
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          >
            star-border
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel=""
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
        }
      }
      testID="non_animatableView"
    >
      <View>
        <View>
          <Text
            size={32}
            style={
              [
                {
                  "color": "#aaaaaa",
                },
                {
                  "paddingLeft": 0,
                  "paddingRight": 8,
                },
                {
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          >
            star-border
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityLabel=""
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
        }
      }
      testID="non_animatableView"
    >
      <View>
        <View>
          <Text
            size={32}
            style={
              [
                {
                  "color": "#aaaaaa",
                },
                {
                  "paddingLeft": 0,
                  "paddingRight": 8,
                },
                {
                  "transform": [
                    {
                      "rotate": "0deg",
                    },
                  ],
                },
              ]
            }
          >
            star-border
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#eb8600",
        "fontFamily": "Roboto",
        "paddingLeft": 8,
        "userSelect": "text",
      }
    }
  />
</View>
`;
